sdk/identity/Azure.Identity/src/IdentityCompatSwitches.cs

```diff
@@ -13,14 +13,16 @@ internal class IdentityCompatSwitches
         internal const string DisableInteractiveThreadpoolExecutionEnvVar = "AZURE_IDENTITY_DISABLE_INTERACTIVEBROWSERTHREADPOOLEXECUTION";
         internal const string DisableCP1ExecutionSwitchName = "Azure.Identity.DisableCP1";
         internal const string DisableCP1ExecutionEnvVar = "AZURE_IDENTITY_DISABLE_CP1";
+        internal const string DisableTenantDiscoverySwitchName = "Azure.Identity.DisableTenantDiscovery";
+        internal const string DisableTenantDiscoveryEnvVar = "AZURE_IDENTITY_DISABLE_TENANTDISCOVERY";
```

Comment:
I'm not sure "tenant discovery" is the right name for this feature. Originally we called this feature multi tenant authentication, as the credentials are not really "discovering" the tenant, but rather just honoring the tenant id passed in as part of the `TokenRequestContext`.